apps/survey/src/components/prompts/standard/SubmitPrompt.vue

```diff
@@ -13,31 +41,193 @@
+import type { PropType } from 'vue';
+import { computed, defineComponent, ref, watch } from 'vue';
 
+import type { Prompt, Prompts } from '@intake24/common/prompts';
+import type { MealState } from '@intake24/common/types';
+import {
+  ReviewMealList,
+  ReviewMealListMobile,
+  ReviewMealListMobileScroll,
+  ReviewMealListScroll,
+} from '@intake24/survey/components/layouts';
 import { usePromptUtils } from '@intake24/survey/composables';
 
 import createBasePrompt from '../createBasePrompt';
 
 export default defineComponent({
   name: 'SubmitPrompt',
 
+  components: {
+    ReviewMealList,
+    ReviewMealListMobile,
+    ReviewMealListScroll,
+    ReviewMealListMobileScroll,
+  },
+
   mixins: [createBasePrompt<'submit-prompt'>()],
 
+  props: {
+    meals: {
+      type: Array as PropType<MealState[]>,
+      required: true,
+    },
+    prompt: {
+      type: Object as PropType<Prompt>,
+      required: true,
+    },
+    review: {
+      type: Object as PropType<Prompts['submit-prompt']['review']>,
+    },
+  },
+
   setup(props, ctx) {
     const { action } = usePromptUtils(props, ctx);
 
-    const isValid = true;
+    const reviewed = ref<string[]>([]);
+
+    const reviewedMobile = ref<string[]>([]);
+
+    const isValid = ref<boolean>(false);
+
+    const isValidMobile = ref<boolean>(false);
+
+    const reachedBottom = ref<boolean>(false);
+
+    const reachedBottomMobile = ref<boolean>(false);
+
+    const showReviewMealListCheckbox = computed(() => {
+      if (
+        props.prompt.component === 'submit-prompt' &&
+        props.prompt.review['desktop'] === 'checkbox'
+      ) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+
+    const showReviewMealListMobileCheckbox = computed(() => {
+      if (
+        props.prompt.component === 'submit-prompt' &&
+        props.prompt.review['mobile'] === 'checkbox'
+      ) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+
+    const showReviewMealListScroll = computed(() => {
+      if (
+        props.prompt.component === 'submit-prompt' &&
+        props.prompt.review['desktop'] === 'scroll'
+      ) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+
+    const showReviewMealListMobileScroll = computed(() => {
+      if (
+        props.prompt.component === 'submit-prompt' &&
+        props.prompt.review['mobile'] === 'scroll'
+      ) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+
+    const updateReviewed = (newReviewed: string[]) => {
+      reviewed.value = newReviewed;
+    };
+
+    const updateReviewedMobile = (newReviewed: string[]) => {
+      reviewedMobile.value = newReviewed;
+    };
+
+    const setReachedBottom = (status: boolean) => {
+      reachedBottom.value = status;
+    };
+
+    const setReachedBottomMobile = (status: boolean) => {
+      reachedBottomMobile.value = status;
+    };
+
+    watch(
+      [
```

Comment:
wonder if we can simplify this, this will be expensive watcher

Reply:
Thanks for the suggestion. The watcher is not necessary and I have changed it to a few computed properties.